### ISSUE TITLE
ValueError: too many values to unpack (expected 3)

### ISSUE THREAD

**Issue by alex** [readthedocs/sphinx_rtd_theme#1343]
```
Traceback (most recent call last):
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/html/__init__.py", line 1096, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/jinja2glue.py", line 194, in render
    return self.environment.get_template(template).render(context)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/themes/basic/page.html", line 10, in top-level template code
    {%- extends "layout.html" %}
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx_rtd_theme/layout.html", line 13, in top-level template code
    {%- set (_ver_major, _ver_minor, _ver_bugfix) = sphinx_version.split('.') | map('int') -%}
ValueError: too many values to unpack (expected 3)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/cmd/build.py", line 279, in build_main
    app.build(args.force_all, filenames)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/application.py", line 350, in build
    self.builder.build_update()
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 302, in build_update
    self.build(to_build,
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 368, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 563, in write
    self._write_serial(sorted(docnames))
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 573, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/html/__init__.py", line 672, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/runner/work/cryptography/cryptography/.tox/docs/lib/python3.10/site-packages/sphinx/builders/html/__init__.py", line 1103, in handle_page
    raise ThemeError(__("An error happened in rendering the page %s.\nReason: %r") %
sphinx.errors.ThemeError: An error happened in rendering the page api-stability.
```

this appears to have been caused by the latest sphinx being version `5.2.0.post0`

https://github.com/pyca/cryptography/actions/runs/3119261059/jobs/5060248154

**Comment by WizardCM** [comment]
Confirmed. Occurs on Sphinx `v5.2.0.post0`

```
writing output... [  2%] backend-design

Traceback (most recent call last):
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 1096, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/jinja2glue.py", line 194, in render
    return self.environment.get_template(template).render(context)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/wizardcm/.local/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/themes/basic/page.html", line 10, in top-level template code
    {%- extends "layout.html" %}
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx_rtd_theme/layout.html", line 13, in top-level template code
    {%- set (_ver_major, _ver_minor, _ver_bugfix) = sphinx_version.split('.') | map('int') -%}
ValueError: too many values to unpack (expected 3)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/cmd/build.py", line 279, in build_main
    app.build(args.force_all, filenames)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/application.py", line 350, in build
    self.builder.build_update()
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 302, in build_update
    self.build(to_build,
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 368, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 563, in write
    self._write_serial(sorted(docnames))
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 573, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 672, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/wizardcm/.local/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 1103, in handle_page
    raise ThemeError(__("An error happened in rendering the page %s.\nReason: %r") %
sphinx.errors.ThemeError: An error happened in rendering the page backend-design.
Reason: ValueError('too many values to unpack (expected 3)')

Theme error:
An error happened in rendering the page backend-design.
Reason: ValueError('too many values to unpack (expected 3)')
```

https://github.com/obsproject/obs-studio/actions/runs/3120097770/jobs/5060455218

**Comment by danielrobbins** [comment]
I have triaged this issue -- please see https://bugs.funtoo.org/browse/FL-10533

The short summary is that the following line in sphinx_rtd_theme assumes that the sphinx version can be split into 3 parts. This is not true with 5.2.0.post0. The relevant line in the traceback is:

  File "/usr/lib/python3.7/site-packages/sphinx_rtd_theme/layout.html", line 13, in top-level template code
    {%- set (_ver_major, _ver_minor, _ver_bugfix) = sphinx_version.split('.') | map('int') -%}
ValueError: too many values to unpack (expected 3)

pypi supports different types of versions besides x.y.z. It is best to parse the version using a more robust version parsing such as packaging.version.Version class.

**Comment by rkdarst** [comment]
Workaround in a project's requirements (as is being done by others in the references above):
```
sphinx!=5.2.0.post0    # remove after next release
```
It would be best if sphinx_rtd_theme releases with the fix in #1344, but another workaround is sphinx releasing with a 3-part version number.  Or every project does the above, which is definitely not good.

**Comment by mara004** [comment]
Frankly, it's a bit hilarious this bug is still labelled with `Needed: replication` 🙃

**Comment by CreatingNull** [comment]
Sphinx is pending a [5.2.1](https://github.com/sphinx-doc/sphinx/issues/10860#issuecomment-1257236167) release, this should resolve the issue (for now).

**Comment by pelson** [comment]
I proposed a fix in #1345.
(posting here for visibility)
